### PR TITLE
Replace URI.escape with ERB::Util.url_encode

### PR DESCRIPTION
**Pull request by elrayle**
This is the closest replacement to what already exists.  There are multiple discussions on what is the correct way to do this.  We may want to revisit later.

See… https://stackoverflow.com/questions/2824126/whats-the-difference-between-uri-escape-and-cgi-escape

This addresses deprecation warnings associated with `URI.escape`

### CODE REVIEW

File: lib/qa/authorities/linked_data/config.rb [hunk 1]
```diff
@@ -18,10 +17,6 @@
 module Qa::Authorities
   module LinkedData
     class Config
-      class << self
```
Comment (elrayle) [inline lib/qa/authorities/linked_data/config.rb]: This should have been removed previously when the use of ERB was moved to another class.

**Comment by elrayle** [comment]
This is related to the work done in PR #196.

**Comment by elrayle** [comment]
Confirmed each modified authority by comparing results before and after for the following URLs...

* http://localhost:3000/qa/search/assign_fast/all?q=periodic+table
* http://localhost:3000/qa/search/crossref/journals?q=chest
* http://localhost:3000/qa/search/geonames?q=port
* http://localhost:3000/qa/search/getty/aat?q=amphora
* http://localhost:3000/qa/search/getty/tgn?q=memphis
* http://localhost:3000/qa/search/getty/ulan?q=washington
* http://localhost:3000/qa/search/loc/subjects?q=History--
* http://localhost:3000/qa/search/linked_data/oclc_fast?q=twain&maximumRecords=10

**Comment by elrayle** [comment]
This replaces PR #196 which made a fix for a single authority.  This makes a consistent fix across all uses of `URI.escape`.

**Comment by elrayle** [comment]
@bbarberBPL As this replaces PR #196, can you review and comment on the changes in this PR?

File: lib/qa/authorities/assign_fast/generic_authority.rb [hunk 2]
```diff
@@ -53,7 +53,7 @@ def build_query_url(q)
       #   See oclc sample code at
       #   http://experimental.worldcat.org/fast/assignfast/js/assignFASTComplete.js
       def clean_query_string(q)
-        URI.escape(q.gsub(/-|\(|\)|:/, ""))
+        ERB::Util.url_encode(q.gsub(/-|\(|\)|:/, ""))
```
Comment (boveus) [inline lib/qa/authorities/assign_fast/generic_authority.rb]: I think `ERB::Util.url_encode` will properly encode all of those except for the dash.  I tested it in IRB really quick
```
2.3.6 :003 > ERB::Util.url_encode("-|\(|\)|:")
 => "-%7C%28%7C%29%7C%3A"
```
I am not really sure  how this is being used elsewhere, so I am not sure what removing the gsub would potentially break.  Allowing the special characters may improve the autosuggest functionality slightly by allowing users to search for terms containing special characters

**Comment by elrayle** [comment]
I tried before and after for...

* http://localhost:3000/qa/search/loc/names?q=Queer%20as%20folk%20(Television%20program%20:%20U.S.)
* http://localhost:3000/qa/search/loc/names?q=King%20Cole%20of%20Spokane,%20Inc.

Both worked in the before and after tests.

**Comment by elrayle** [comment]
@hackmastera You appear to be the original committer for assign_fast.  There is a question about the clean_query_string method for assign_fast.  The call to `URI.escape` is being replaced by `ERB::Util.url_encode`.  The original  code includes a gsub that appears to remove some characters.  With `ERB::Util.url_encode`, these characters appear to be encoded properly.  I am wondering two things...

* Is the new approach adequate and the gsub can be removed?
* Do you have an example that requires the gsub that can be used for before and after testing?

**Comment by hackmastera** [comment]
@elrayle It looks like `assign_fast` is still in use in https://github.com/sciencehistory/chf-sufia/blob/master/app/assets/javascripts/edit_metadata.js; maybe @jrochkind or @eddierubeiz could identify some tests?

My vague memory is that the characters I removed were an issue on the API side, rather than a matter of encoding.

**Pull request merged**